src/include/platform/internal/GenericPlatformManagerImpl_FreeRTOS.ipp

```diff
@@ -277,10 +266,13 @@ template <class ImplClass>
 void GenericPlatformManagerImpl_FreeRTOS<ImplClass>::EventLoopTaskMain(void * arg)
 {
     ChipLogDetail(DeviceLayer, "CHIP event task running");
-    static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->Impl()->RunEventLoop();
-    // TODO: At this point, should we not
-    // vTaskDelete(static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->mEventLoopTask)?
-    // Or somehow get our caller to do it once this thread is joined?
+    GenericPlatformManagerImpl_FreeRTOS<ImplClass> * platformManager =
+        static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg);
+    platformManager->Impl()->RunEventLoop();
+    vTaskDelete(NULL);
+    vQueueDelete(platformManager->mChipEventQueue);
```

Comment:
Should we delete the Event Queue in the `_Shutdown()` function? As the user might call the `StopEventLoopTask()` and  start the task again. In this case, the event queue might not be initialized again.

Reply:
Sure. Done